__test__/integration/printquestion.test.js

```diff
@@ -0,0 +1,55 @@
+const Helper = require('hubot-test-helper');
+
+const { SubmitAnswerMock } = require('./util');
+
+const timeStamp = Date.now();
+
+const UP_TIME = 11 * 47 * 1000;
+
+const NOW_PLUS_UP_TIME = new Date(timeStamp + UP_TIME);
+
+const UP_TIME_MINUTES = Math.floor(UP_TIME / 60 / 1000);
+
+const UP_TIME_STRING = `~${UP_TIME_MINUTES} minutes`;
+
+jest.useFakeTimers();
+
+describe('When a user types \'!printquestion\'', () => {
+    let lqBotRoom;
+    let submitAnswerMock;
+    beforeEach(() => {
+        const helper = new Helper('../../src/loaded-questions.js');
+        lqBotRoom = helper.createRoom({ httpd: false, name: '#random' });
+        submitAnswerMock = new SubmitAnswerMock(lqBotRoom);
+    });
+    test('If there\'s no active question, respond with a message to that effect', async () => {
+        await lqBotRoom.user.say('billy', '!printquestion');
+        expect(lqBotRoom.messages).toHaveLength(2);
+        expect(lqBotRoom.messages[1]).toHaveLength(2);
+        expect(lqBotRoom.messages[1][1]).toEqual(expect.stringContaining('There isn\'t a question loaded right now'));
+    });
+    describe('If there\'s an active question, respond with a message...', () => {
+        beforeEach(async () => {
+            const helper = new Helper('../../src/loaded-questions.js');
+            lqBotRoom = helper.createRoom({ httpd: false, name: '#random' });
+            await lqBotRoom.user.say('billy', '!loadquestion');
+            await submitAnswerMock.sendPrivate('billy', 'submit answer this is billy\'s answer');
+            await submitAnswerMock.sendPrivate('bobby', 'submit answer this is bobby\'s answer');
+            Date = jest.fn(() => NOW_PLUS_UP_TIME);
+            await lqBotRoom.user.say('billy', '!printquestion');
+        });
+        afterEach(async () => {
+            await lqBotRoom.user.say('billy', '!endquestion');
+        });
+        test('...that contains the up time of the question', () => {
+            expect(lqBotRoom.messages).toHaveLength(6);
+            expect(lqBotRoom.messages[5]).toHaveLength(2);
+            expect(lqBotRoom.messages[5][1]).toEqual(expect.stringContaining(UP_TIME_STRING));
+        });
+        test('...that contains the number of answers submitted', () => {
+            expect(lqBotRoom.messages).toHaveLength(6);
+            expect(lqBotRoom.messages[5]).toHaveLength(2);
+            expect(lqBotRoom.messages[5][1]).toEqual(expect.stringContaining('2'));

```

Comment:
Since this is testing the number of answers, would it be a good idea to have the `stringContaining` test for '2 answers'? I'm not sure if the `[5][1]` indexing is selecting a specific line or not.

Reply:
lqBotRoom.messages is where the test bot helper keeps the messages for a room. It ends up taking the shape of [[userName, messageText], ...]. So the indexing is derived from me manually counting how many messages there should be between the users and the bot, and then finding the last message sent. So it is targeting a specific message, but I think adding 'answers' would be a good idea just to make it more explicit. 

I'm unsure how to make the indexing more clear. I suppose an inline comment in the code could accomplish that. At least to explain the composition of the messages array.